### PR TITLE
Refactors the bulk string framework

### PR DESCRIPTION
**Pull request by optimumtact**
```
In favour of a better json based solution
This overhauls:
    -AI ion laws
    -NPC chatter text

The system is extended to support further splitting of a chosen line
where a line is picked and then searched for subkey's in the form of
@pick(\D+) - sub keys are then picked from the appropriate list that
matches the \D+ key and replaced over top of the placeholder

This is used to add braindamage gibbering messages to the system
```

This PR is a WIP and will not be finished just yet, so do not merge

### CODE REVIEW

File: code/__HELPERS/_string_lists.dm [hunk 1]
```diff
@@ -1,27 +1,46 @@
...
+		var/start = findtext('@pick(')
+		var/end = 0
+		var/sub_key = ''
+		var/lookup_val = ''
+		while(start)
+			end = findtext(')', start)
+			sub_key = copytext(response, start+6, end)
+			lookup_val = pick_list(filename, sub_key)
+			world << lookup_val
+			start = findtext('@pick(')
```
Comment (optimumtact) [inline code/__HELPERS/_string_lists.dm]: Are people of the opinion that regex is faster than straight findtext hacks like this? Have any performance tests been done?

**Comment by GunHog** [comment]
PR with title "Removes the AI", made by Oranges, red "x" from failed travis. 

You are such a troll.

File: code/__HELPERS/_string_lists.dm [hunk 2]
```diff
@@ -1,27 +1,40 @@
...
+//This is a hack to track which file is being processed for string subkey
+//replacements, yes it's ugly, I don't know an easier way though
+var/global/string_filename_current_key
+
+/proc/strings_replacement(filename, key)
+	load_strings_file(filename)
+
+	if((filename in string_cache) && (key in string_cache[filename]))
+		var/response = pick(string_cache[filename][key])
+		var/regex/r = regex('@pick\\((\\D+)\\)', 'g')
```
Comment (optimumtact) [inline code/__HELPERS/_string_lists.dm]: this regex is equivalent to 
    @pick(\D+)
with the \D+ being captured (any non whitespace or digit symbol)

File: code/__HELPERS/_string_lists.dm [hunk 3]
```diff
@@ -1,27 +1,40 @@
 #define pick_list(FILE, KEY) (pick(strings(FILE, KEY)))
+#define pick_list_replacements(FILE, KEY) (strings_replacement(FILE, KEY))
 
 var/global/list/string_cache
+//This is a hack to track which file is being processed for string subkey
+//replacements, yes it's ugly, I don't know an easier way though
+var/global/string_filename_current_key
```
Comment (optimumtact) [inline code/__HELPERS/_string_lists.dm]: This is an ugly hack to make up for being unable to pass the filename through to the string_subkey_lookup function

if anyone knows a better way I'm open to it.

Comment (RemieRichards) [inline code/__HELPERS/_string_lists.dm]: Why cant you pass filenames?
you can totally pass filenames.

Comment (optimumtact) [inline code/__HELPERS/_string_lists.dm]: I don't have control over how the string_subkey_lookup is called, it's done by the regex datum's Replace method.

Comment (optimumtact) [inline code/__HELPERS/_string_lists.dm]: I think I have an idea of how to approach this

**Comment by AnturK** [comment]
Neat. Mind checking how hard would it be to transfer AI laws to json ? So we can have them fully generic instead of few hardcoded ones.

**Comment by optimumtact** [comment]
I could look at that for sure

**Comment by optimumtact** [comment]
This is not dead, just didn't have much spare time last week!

**Comment by optimumtact** [comment]
I'm going to squash and then this will be ready to go!

If merged I'll move AI laws to it.

**Comment by optimumtact** [comment]
Actually one more thing I want to test sorry

will let you know

**Comment by optimumtact** [comment]
Running into a byond bug with this at the moment

http://www.byond.com/forum/?post=2049269

**Comment by optimumtact** [comment]
This is ready to go once it passes travis.

**Pull request merged**